### ISSUE TITLE
--mobile flag

### ISSUE THREAD

**Issue by maytis** [angular/mobile-toolkit#108]
I'm using the angular-cli: 1.0.0-beta.16. Mobile flag is still disabled. When can we expect the support for mobile flag.

**Comment by ruedagato** [comment]
+1

**Comment by chimon2000** [comment]
@maytis I have found that `ng init --mobile` works if you've already created the project folder.

**Comment by friedger** [comment]
Don't try ng init --mobile
https://github.com/angular/angular-cli/issues/2679

**Comment by michaeljota** [comment]
@friedger that issue appears to be fixed and closed. Have not test it.

**Comment by friedger** [comment]
@michaeljota The resolution of the issue that an error is thrown if you use the --mobile flag

**Comment by michaeljota** [comment]
@friedger Oh! Thanks...

**Comment by agalazis** [comment]
still not enabled although the cause was fixed(universal uses 2.0.0 now)

**Comment by elekzalan** [comment]
@agalazis Moreover, universal supports aot compilation since it's 2.1.0-rc.1 release. The `--mobile` flag is a must-have for a bunch of projects...

**Comment by born2net** [comment]
its broken due to the fact universal is being added with modules.
however, I really don't like universal, I wish we could have an AOT version of this project (no universal)

**Comment by iansim** [comment]
without --mobile does angular worth to learn?

**Comment by worldspawn** [comment]
Checked on this mobile stuff like a year ago and it had a great bug (something to do with universal) that produced blank templates or a template with the word "null" in it. A year later and now its just disabled, effectively disabling this entire project?!? Ng eco-system feels like an endless chore of beta-never-gets-released-ware. I know, i know, its open source free ware but shit, i'm going to whinge anyway :P

**Comment by webmaxru** [comment]
CLI and `--mobile` are just helpers. You can use Mobile Toolkit separately: https://github.com/angular/mobile-toolkit/issues/115#issuecomment-262467713

**Comment by gdi2290** [comment]
here's a repo with module packages https://github.com/webmaxru/pwa-guide-ngpoland/tree/step-dashboard

**Comment by sebastianovide** [comment]
i've just started a PWA with angular2. In the https://angular.io/ there is a big invitation to try @angular/material. https://material.angular.io/guide/getting-started instructs to use angular-cli but https://github.com/angular/angular-cli has disabled --mobile. Also  https://mobile.angular.io/ indicates to use angular-cli.

The previous link is a nice demo done by @gdi2290 (thanks) but as it is not using the tools suggested by the official documentation, using it as a starting point seems risky. Where are we going to get support from ?

mobile-toolkit seems very quiet (https://github.com/angular/mobile-toolkit/pulse/monthly) and angular-cli position about --mobile is not clear https://github.com/angular/angular-cli/issues/2228

So starting today a new PWA using angular2 seems very painful as there is not official working documentation on how to do it. 

Should we hold before using angular2 for PWA ?

**Comment by hansl** [comment]
@alxhub It would be nice to remove any mention of the cli and `--mobile` from the microsite. I think this issue can be either replaced by a new issue for removing the tutorial, or considered for tracking removing the tutorial.

**Comment by Charlkie14** [comment]
@sebastianovide 
> Should we hold before using angular2 for PWA ?

I'm still wondering if we should wait on creating pwa with Angular?

**Comment by friedger** [comment]
@Charlkie14 Definitively, build PWAs with Angular, this should not depend on the tools!

**Comment by Charlkie14** [comment]
Yes your right @friedger. :) I'm trying it out now.

**Comment by sebastianovide** [comment]
@Charlkie14  I'm using angular for PWA. Check out ionic2.

**Comment by cre8** [comment]
@Charlkie14 ionic2 is similar to PWAs, but it's not the same. Ionic is building an app for the smartphones, PWAs don't need to be packed up to work on different mobile platforms.

**Comment by sebastianovide** [comment]
@cre8. Actually I'm building a PWA with Ionic2. Pure PWA with service woker without Phonegap. In us part of ionic2 out of the box.

**Comment by jakeNiemiec** [comment]
@sebastianovide very good to know. Can ionic2 do AoT now?

**Comment by mgechev** [comment]
I believe we can close this issue. Here's the PR by @alxhub https://github.com/angular/angular-cli/commit/cb2e418 to the CLI.
